corehq/apps/users/models.py

```diff
@@ -562,8 +562,8 @@ def add_as_web_user(self, domain, role, primary_location_id=None,
             self.get_domain_membership(domain).program_id = program_id
         if domain_obj.uses_locations:
             if primary_location_id:
-                self.set_location(domain, primary_location_id)
-            self.reset_locations(domain, assigned_location_ids)
+                self.set_location(domain, primary_location_id, commit=False)
+            self.reset_locations(domain, assigned_location_ids, commit=False)
```

Comment:
I looked and it doesn't seem to the case. But just noting that we need to make sure to check there isn't temporal coupling that requires location to be committed here

Reply:
just looking at this comment for `reset_locations` 
```            
This should be called after updating primary location via set_location/unset_location
If primary-location is not set, then next available location from
assigned_location_ids is set as the primary-location
```
I thought there was a possibility that it requires `location_id` to be saved first. but doesn't seem like it